test/jdk/javax/swing/JFileChooser/4400728/JFileChooserDefaultDirectoryTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2004, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.File;
+import javax.swing.JFileChooser;
+
+/*
+ * @test
+ * @bug 4400728
+ * @summary This testcase tests CCC4400728 request, checks whether JFileChooser
+ *          constructor set correct default directory or not, it is typically
+ *          the "My Documents" folder on Windows, and the user's home directory
+ *          on Unix.
+ * @run main JFileChooserDefaultDirectoryTest
+ */
+public class JFileChooserDefaultDirectoryTest {
+
+    public static void main(String[] args) {
+        JFileChooser jFileChooser = new JFileChooser();
```

Comment:
Please make sure that the Swing will be used on EDT.

Reply:
Done